src/bugjira/field_data_generator.py

```diff
@@ -0,0 +1,165 @@
+import abc
+import json
+from typing import List
+
+from pydantic import BaseModel, ConfigDict, ValidationError
+from stevedore import driver
+
+from bugjira.common import BUGZILLA, JIRA, PLUGIN_NAMESPACE
+from bugjira.exceptions import FieldDataGeneratorException
+from bugjira.field import BugzillaField, JiraField
+
+
+class ValidFieldData(BaseModel):
+    """This class defines the valid format for the json data loaded by the
+    FieldDataGenerator classes in this module
+    """
+    model_config = ConfigDict(extra="forbid")
+    bugzilla_field_data: List[BugzillaField]
+    jira_field_data: List[JiraField]
+
+
+class FieldDataGeneratorInterface(metaclass=abc.ABCMeta):
+    """This abstract base class represents the interface that
+    FieldDataGenerator classes should implement. Plugin classes for generating
+    field data should implement the get_field_data method.
+    """
+
+    @abc.abstractmethod
+    def __init__(self, config):
+        """Init method"""
+
+    @abc.abstractmethod
+    def get_field_data(self) -> List:
+        """Return the field data as an Iterable
+
+        :return: List of raw field data to be used to instantiate subclasses of
+            the BugjiraField class
+        :rtype: List
+        """
+
+
+class FieldDataGenerator(FieldDataGeneratorInterface):
+    """This is the superclass for the default/provided FieldDataGenerator
+    plugin classes, BugzillaFieldDataGenerator and JiraFieldDataGenerator.
+    Both subclasses use the same sample json field data file, so we load that
+    data in the superclass's __init__ method.
+    """
+
+    def __init__(self, config):
+        self.field_data = {}
+        if config:
+            field_data_path = config.get("field_data_path", "")
+            if field_data_path:
+                with open(field_data_path, "r") as file:
+                    field_data = json.load(file)
+                    try:
+                        # use pydantic class to validate the input data
+                        ValidFieldData(**field_data)
+                    except ValidationError:
+                        raise FieldDataGeneratorException(
+                            "Invalid field data detected"
```

Comment:
Good point. In the revision I just sent up, I've chained the exceptions so that a caller can access the ValidationError via the dunder-"cause" attribute of the FieldDataGeneratorException. I think it is probably better to let that bubble up to the client and then we can do a human-readable formatting of the ValidationError at that point.